src/react/projects/spark-core-react/src/SprkMasthead/components/SprkMastheadSelector/SprkMastheadSelector.js

```diff
@@ -0,0 +1,233 @@
+/* global window */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { uniqueId } from 'lodash';
+import SprkIcon from '../../../SprkIcon/SprkIcon';
+
+class SprkMastheadSelector extends Component {
+  constructor(props) {
+    super();
+    this.state = {
+      isOpen: false,
+      triggerText: props.defaultTriggerText,
+      choiceItems: props.choices.items.map(item => ({ id: uniqueId(), ...item })),
+    };
+    this.openDropdown = this.openDropdown.bind(this);
+    this.closeDropdown = this.closeDropdown.bind(this);
+    this.updateTriggerText = this.updateTriggerText.bind(this);
+    this.closeOnEsc = this.closeOnEsc.bind(this);
+    this.closeOnClickOutside = this.closeOnClickOutside.bind(this);
+    this.myRef = React.createRef();
+  }
+
+  componentDidMount() {
+    window.addEventListener('keydown', this.closeOnEsc);
+    window.addEventListener('focusin', this.closeOnClickOutside);
+    window.addEventListener('click', this.closeOnClickOutside);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('keydown', this.closeOnEsc);
+    window.removeEventListener('focusin', this.closeOnClickOutside);
+    window.removeEventListener('click', this.closeOnClickOutside);
+  }
+
+  closeOnEsc(e) {
+    if (e.key === 'Escape') {
+      this.closeDropdown();
+    }
+  }
+
+  closeOnClickOutside(e) {
+    if (!this.myRef.current.contains(e.target)) {
+      this.closeDropdown();
+    }
+  }
+
+  openDropdown() {
+    this.setState({
+      isOpen: true,
+    });
+  }
+
+  closeDropdown() {
+    this.setState({
+      isOpen: false,
+    });
+  }
+
+  updateTriggerText(text) {
+    this.setState({
+      triggerText: text,
+    });
+  }
+
+  render() {
+    const {
+      additionalClasses,
+      additionalIconClasses,
+      additionalTriggerClasses,
+      additionalTriggerTextClasses,
+      analyticsString,
+      choices,
+      iconType,
+      idString,
+      isFlush,
+    } = this.props;
+    const { choiceFunction, footer } = choices;
+    const { choiceItems, isOpen, triggerText } = this.state;
+
+    return (
+      // eslint-disable-next-line
+      <div
+        role="dialog"
+        ref={this.myRef}
+        className={classNames({ 'sprk-c-MastheadMask': isOpen && isFlush })}
+        onClick={() => { if (isOpen) { this.closeDropdown(); } }}
+      >
+        <div className={classNames({ 'sprk-o-Box': isFlush })}>
+          <a
+            className={classNames(
+              'sprk-c-Masthead__selector sprk-b-Link sprk-b-Link--plain sprk-o-Stack sprk-o-Stack--split@xxs sprk-o-Stack--center-column',
+              additionalTriggerClasses,
+            )}
+            data-analytics={analyticsString || 'undefined'}
+            data-id={idString || 'undefined'}
```

Comment:
You don't need the logical OR here because if the idString prop is not defined then React will not render out the data-id tag.
```suggestion
            data-id={idString}
```